src/main/groovy/spock/genesis/generators/composites/PermutationGenerator.groovy

```diff
@@ -0,0 +1,97 @@
+package spock.genesis.generators.composites
+
+import groovy.transform.CompileStatic
+import spock.genesis.generators.Generator
+import spock.genesis.generators.UnmodifiableIterator
+
+@CompileStatic
+class PermutationGenerator<E> extends TupleGenerator<E> {
+
+    static final int MAX_PERMUTATIONS = 10000
+    private final int maxDepth
+
+    PermutationGenerator(List<Generator<E>> generators, int maxDepth) {
+        super(generators)
+        this.maxDepth = maxDepth
+    }
+
+    PermutationGenerator(List<Generator<E>> generators) {
+        super(generators)
+        this.maxDepth = pickMaxDepth(generators)
+    }
+
+    @Override
+    UnmodifiableIterator<List<E>> iterator() {
+        final List<UnmodifiableIterator<E>> ITERATORS = generators.collect { generator ->
+            generator.take(maxDepth).iterator()
+        }
+
+        new UnmodifiableIterator<List<E>>() {
+            private final List<List<E>> emitted = []
+            private final List<List<E>> currentEmmited = []
+            private int column = 0
+            private E value
+            private Iterator<List<E>> emittedIterator
+
+            @Override
+            boolean hasNext() {
+                if (emitted == []) {
```

Comment:
In order to keep symmetry with line `61` and still using idiomatic Groovy, I would have used `!emmited`.